src/rgw/rgw_tag.cc

```diff
@@ -35,6 +35,9 @@ int RGWObjTags::check_and_add_tag(const string&key, const string& val){
 }
 
 int RGWObjTags::set_from_string(const string& input){
+  if (input.empty()) {
+    return 0;
+  }
```

Comment:
wouldn't `boost::split()` just give us an empty vector, and we'd return success anyway?

Reply:
no, it gives us a vector that contains an empty string.

Reply:
In that case, maybe the simplest change would be to have check_and_add_tag() return success for empty key?

Reply:
An empty tagset is well-formed, but an empty key is not.